xray/utils/resultstable.go

```diff
@@ -873,31 +899,123 @@ func GetUniqueKey(vulnerableDependency, vulnerableVersion, xrayID string, fixVer
+			var applicableDetails *formats.Applicability
+			if details, exists := applicabilityMap[relatedCve]; exists {
+				applicableDetails = details
+			} else {
+				applicableDetails = &formats.Applicability{
+					Status:             isApplicableResult(contexualAnalysisResult),
+					ScannerDescription: searchTargetData[relatedCve],
+				}
+				applicabilityMap[relatedCve] = applicableDetails
+			}
+			// Add new evidences
+			for l := range contexualAnalysisResult.Locations {
+				location := contexualAnalysisResult.Locations[l]
+				applicableDetails.Evidence = append(applicableDetails.Evidence, formats.Evidence{
+					SourceCodeLocationRow: formats.SourceCodeLocationRow{
+						File:       GetLocationFileName(location),
+						LineColumn: GetStartLocationInFile(location),
+						Snippet:    GetLocationSnippet(location),
+					},
+					Reason: GetResultMsgText(contexualAnalysisResult),
+				})
+			}
+		}
+	}
+
+	return &applicabilityMap
+}
+
+func convertCves(cves []services.Cve) []formats.CveRow {
+	var cveRows []formats.CveRow
+	for _, cveObj := range cves {
+		cveRows = append(cveRows, formats.CveRow{Id: cveObj.Id, CvssV2: cveObj.CvssV2Score, CvssV3: cveObj.CvssV3Score})
+	}
+	return cveRows
+}
+
 // If at least one cve is applicable - final value is applicable
 // Else if at least one cve is undetermined - final value is undetermined
 // Else (case when all cves aren't applicable) -> final value is not applicable
-func getApplicableCveValue(extendedResults *ExtendedScanResults, xrayCves []formats.CveRow) ApplicabilityStatus {
+func extractApplicabilityValues(extendedResults *ExtendedScanResults, cveRows []formats.CveRow) (value ApplicabilityStatus) {
 	if !extendedResults.EntitledForJas || len(extendedResults.ApplicabilityScanResults) == 0 {
 		return NotScanned
 	}
-
-	if len(xrayCves) == 0 {
+	if len(cveRows) == 0 {
 		return ApplicabilityUndetermined
 	}
-	cveExistsInResult := false
-	finalApplicableValue := NotApplicable
-	for _, cve := range xrayCves {
-		if currentCveApplicableValue, exists := extendedResults.ApplicabilityScanResults[cve.Id]; exists {
-			cveExistsInResult = true
-			if currentCveApplicableValue == Applicable {
-				return currentCveApplicableValue
-			} else if currentCveApplicableValue == ApplicabilityUndetermined {
-				finalApplicableValue = currentCveApplicableValue
+	// Entitled for Jas and have at least one cve.
+	// Only if we don't find any undetermined or applicable it will stay the default 3rd priority Not applicable value
+	value = NotApplicable
+	for _, cve := range cveRows {
+		applicabilityStatus := attachCveApplicabilityDetails(cve, extendedResults.ApplicabilityScanResults)
+		if applicabilityStatus == Applicable {
+			// Found at least one applicable - 1st priority
+			value = Applicable
+		} else if applicabilityStatus == ApplicabilityUndetermined && value == NotApplicable {
+			// Found at lest one undetermined and no applicable - 2nd priority
+			value = ApplicabilityUndetermined
+		}
+	}
+	return
```

Comment:
not the same functionality as before, lets leave this function as is and just change the fetch function